app-misc/figlet/figlet-222.recipe

```diff
@@ -9,31 +9,41 @@ COPYRIGHT="1991, 1993, 1994 Glenn Chappell and Ian Chai
 	2002 Christiaan Keet"
 LICENSE="AFL v2.1"
 REVISION="1"
-SOURCE_URI="ftp://ftp.figlet.org/pub/figlet/program/unix/figlet222.tar.gz"
-CHECKSUM_MD5="5f84ad52d092d5db4ad5916df767646b"
-STATUS_HAIKU="stable"
-DEPEND=""
+SOURCE_URI="http://ftp.nsysu.edu.tw/FreeBSD/ports/local-distfiles/obrien/figlet222.tar.gz"
+CHECKSUM_SHA256="21f8f9e9b2190fbc0f801fb4704355e5d73d36c55670272cd08cad3408d1f5fe"
+SOURCE_DIR="figlet222"
+
+ARCHITECTURES="x86_gcc2 x86 x86_64"
+
+PROVIDES="
+	figlet = $portVersion
+	cmd:figlet = $portVersion
+	cmd:chkfont = $portVersion
+	cmd:figlist = $portVersion
+	cmd:showfigfonts = $portVersion
+	"
+REQUIRES="
+	haiku
+	"
+
+BUILD_REQUIRES="
+	haiku_devel
+	"
+BUILD_PREREQUIRES="
+	cmd:gcc
+	cmd:make
+	"
+
+fontFolder=$dataDir/figlet/fonts
 
 BUILD()
 {
-	cd figlet222
-	make
+	mkdir -p $binDir $manDir/man6 $fontFolder
+	make DESTDIR=$binDir MANDIR=$manDir/man6 DEFAULTFONTDIR=$fontFolder
 }
 
 INSTALL()
 {
-	cd figlet222
-	DISTRODIR=${DESTDIR}
-	MANDIR=${DISTRODIR}/`finddir B_APPS_DIRECTORY`/figlet/share/man
-	DEFAULTFONTDIR=${DISTRODIR}/`finddir B_APPS_DIRECTORY`/figlet/fonts
-	DESTDIR=${DISTRODIR}/`finddir B_APPS_DIRECTORY`/figlet
-	mkdir -p ${DESTDIR}/tmp
-	mkdir -p ${MANDIR}/man6
-	mkdir -p ${DEFAULTFONTDIR}/tmp
-	make install DESTDIR=${DESTDIR} MANDIR=${MANDIR} \
-		DEFAULTFONTDIR=${DEFAULTFONTDIR}
-	chmod 755 ${DESTDIR}/showfigfonts
-	chmod 755 ${DESTDIR}/figlist
-	rm -rf ${DESTDIR}/tmp
-	rm -rf ${DEFAULTFONTDIR}/tmp
+	make install DESTDIR=$binDir MANDIR=$manDir/man6 DEFAULTFONTDIR=$fontFolder
+	chmod +x $binDir/chkfont $binDir/figlist $binDir/showfigfonts  
```

Comment:
@Ace002, when you remove the two white spaces at the end of the line with the call to chmod, don't forget to also skip the $binDir/chkfont (for the chmod) as this is a compiled runtime (and gcc already does the chmod for us.)